src/portable/synopsys/dwc2/dcd_dwc2.c

```diff
@@ -630,6 +630,27 @@ void dcd_sof_enable(uint8_t rhport, bool en)
 /* DCD Endpoint port
  *------------------------------------------------------------------*/
 
+// Check is IN/OUT endpoint is avaliable before opening it
+static bool dcd_edpt_check_if_avaliable(uint8_t rhport, uint8_t direction)
```

Comment:
Would it make sense to wrap this function in a `#if TU_CHECK_MCU(OPT_MCU_ESP32S2, OPT_MCU_ESP32S3, OPT_MCU_ESP32P4)` block since it's not applicable to other MCUs with a DWC controller?